### DISCUSSION TITLE
Higher order interpolation on regular grids

### DISCUSSION THREAD

**Discussion by flabowski** [mathLab/EZyRB#245] [Ideas]
**Is your feature request related to a problem? Please describe.**
I have my data sampled on a regular grid in the parameter space.

**Describe the solution you'd like**
I am using a regular grid interpolation, see [RegularGridInterpolator](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.RegularGridInterpolator.html): `Interpolation on a regular or rectilinear grid in arbitrary dimensions. Supported are “linear”, “nearest”, “slinear”, “cubic”, “quintic” and “pchip”`. It would be nice to have this available through EZyRB directly.

**Describe alternatives you've considered**
The [LinearNDInterpolator](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.LinearNDInterpolator.html) which is wrapped [here](https://github.com/mathLab/EZyRB/blob/master/ezyrb/linear.py) is **much slower**. 
For 1D data, the [interp1d class](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html) is used, but in EZyRB it allows for the default linear interpolation only.

**Additional context**
#166 suggested new wrappers. Beside the RBF interpolation, I found that the [LinearNDInterpolator](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.LinearNDInterpolator.html) is wrapped [here](https://github.com/mathLab/EZyRB/blob/master/ezyrb/linear.py). 
It does not perform well, because the interpolant is constructed by triangulating the input data with Qhull. Moreover, it allows only for linear interpolation. The RegularGridInterpolator takes advantage of the grid structure and allows for higher order interpolation (scipy 1.10 or higher required).

**Comment by mtezzele** [comment]
Hi @flabowski! I think this could be a great addition to the package. We just need to make everything compatible with the ROM class. I see you opened a PR. I will leave some comments there.
